plugins/source/hubspot/resources/plugin/client.go

```diff
@@ -0,0 +1,156 @@
+}
+
+func (c *Client) Sync(ctx context.Context, options plugin.SyncOptions, res chan<- message.SyncMessage) error {
+	if c.options.NoConnection {
+		return fmt.Errorf("no connection")
+	}
+
+	tables, err := c.allTables.FilterDfs(options.Tables, options.SkipTables, options.SkipDependentTables)
+	if err != nil {
+		return err
+	}
+
+	var stateClient state.Client
+	if options.BackendOptions == nil {
+		c.logger.Info().Msg("No backend options provided, using no state backend")
+		stateClient = &state.NoOpClient{}
+		c.backendConn = nil
+	} else {
+		c.backendConn, err = grpc.DialContext(ctx, options.BackendOptions.Connection,
+			grpc.WithTransportCredentials(insecure.NewCredentials()),
+			grpc.WithDefaultCallOptions(
+				grpc.MaxCallRecvMsgSize(maxMsgSize),
+				grpc.MaxCallSendMsgSize(maxMsgSize),
+			),
+		)
+		if err != nil {
+			return fmt.Errorf("failed to dial grpc source plugin at %s: %w", options.BackendOptions.Connection, err)
+		}
+		stateClient, err = state.NewClient(ctx, c.backendConn, options.BackendOptions.TableName)
+		if err != nil {
+			return fmt.Errorf("failed to create state client: %w", err)
+		}
+		c.logger.Info().Str("table_name", options.BackendOptions.TableName).Msg("Connected to state backend")
+	}
+
+	schedulerClient, err := client.New(ctx, c.logger, c.config, stateClient)
+	if err != nil {
+		return err
+	}
+
+	syncErr := c.scheduler.Sync(ctx, schedulerClient, tables, res, scheduler.WithSyncDeterministicCQID(options.DeterministicCQID))
+
+	if err = stateClient.Flush(ctx); err != nil {
+		return errors.Join(fmt.Errorf("failed to save state: %w", err), syncErr)
+	}
+
+	return syncErr
+}
+
+func Configure(_ context.Context, logger zerolog.Logger, specBytes []byte, options plugin.NewClientOptions) (plugin.Client, error) {
+	c := &Client{
+		logger:    logger,
+		options:   options,
+		allTables: getTables(),
+	}
+	if options.NoConnection {
+		return c, nil
+	}
+	s := spec.Spec{}
+	if err := json.Unmarshal(specBytes, &s); err != nil {
+		return nil, err
+	}
+	s.SetDefaults()
+	if err := s.Validate(); err != nil {
+		return nil, err
+	}
+	c.config = s
+	c.scheduler = scheduler.NewScheduler(scheduler.WithLogger(logger), scheduler.WithConcurrency(s.Concurrency))
+	return c, nil
+}
+
+func getTables() schema.Tables {
+	tables := schema.Tables{
+		crm.Contacts(),
+		crm.Companies(),
+		crm.Deals(),
+		crm.LineItems(),
+		crm.Products(),
+		crm.Tickets(),
+		crm.Quotes(),
+		crm.Owners(),
+		crm.Pipelines(),
+	}
+	if err := transformers.TransformTables(tables); err != nil {
+		panic(err)
+	}
+	if err := transformers.Apply(tables, titleTransformer); err != nil {
+		panic(err)
+	}
+	for _, table := range tables {
+		schema.AddCqIDs(table)
+	}
+	return tables
+}
+
+func titleTransformer(table *schema.Table) error {
+	if table.Title != "" {
+		return nil
+	}
+	exceptions := maps.Clone(docs.DefaultTitleExceptions)
+	for k, v := range customExceptions {
+		exceptions[k] = v
+	}
+	csr := caser.New(caser.WithCustomExceptions(exceptions))
```

Comment:
extract to init only once

Reply:
Done in [602cc6c](https://github.com/cloudquery/cloudquery/pull/17470/commits/602cc6c03fc88a6d35901a8b83d811dc1bf77e1f)